restapi/src/main/java/io/stargate/web/resources/v2/schemas/UserDefinedTypesResource.java

```diff
@@ -0,0 +1,388 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.web.resources.v2.schemas;
+
+import static io.stargate.web.docsapi.resources.RequestToHeadersMapper.getAllHeaders;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+import com.codahale.metrics.annotation.Timed;
+import com.google.common.base.Strings;
+import io.stargate.auth.Scope;
+import io.stargate.auth.SourceAPI;
+import io.stargate.auth.entity.ResourceKind;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.Column.Kind;
+import io.stargate.db.schema.Column.Type;
+import io.stargate.db.schema.ImmutableUserDefinedType;
+import io.stargate.db.schema.Keyspace;
+import io.stargate.db.schema.Table;
+import io.stargate.db.schema.UserDefinedType;
+import io.stargate.web.models.Error;
+import io.stargate.web.models.ResponseWrapper;
+import io.stargate.web.models.UserDefinedTypeAdd;
+import io.stargate.web.models.UserDefinedTypeField;
+import io.stargate.web.models.UserDefinedTypeResponse;
+import io.stargate.web.resources.AuthenticatedDB;
+import io.stargate.web.resources.Converters;
+import io.stargate.web.resources.Db;
+import io.stargate.web.resources.RequestHandler;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.apache.cassandra.stargate.db.ConsistencyLevel;
+
+/**
+ * Exposes REST Endpoint to work with Cassandra User Defined Types
+ *
+ * @see https://cassandra.apache.org/doc/latest/cql/types.html
+ */
+@Api(
+    produces = APPLICATION_JSON,
+    consumes = APPLICATION_JSON,
+    tags = {"schemas"})
+@Produces(APPLICATION_JSON)
+@Consumes(APPLICATION_JSON)
+@Path("/v2/schemas/keyspaces/{keyspaceName}/types")
+public class UserDefinedTypesResource {
```

Comment:
I see create, read, and delete. Is update a thing for UDTs?

Reply:
Good question. It looks like it can be updated, adding and removing fields.

Reply:
According to https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/cqlAlterType.html the two update operations available to UDT are 1) add new columns and 2) rename columns. The former is already implemented in QueryBuilderImpl (but with a bug in update type query building that I fixed) while the latter is not. Should we allow this?

Reply:
It looks like there's 3?
![Uploading Screen Shot 2021-05-05 at 5.15.45 PM.png…]()

Reply:
Oops, sorry. Is it worth including these two other operations in this PR? Can it be done on a follow up PR?

Reply:
Follow up is fine